hostfactory/host_provider/src/cyclecloud_provider.py

```diff
@@ -41,315 +41,75 @@ class InvalidCycleCloudVersionError(RuntimeError):
+    def templates(self): 
         try:
-            symphony_templates = self._update_templates()
-            return self.json_writer({"templates": symphony_templates, "message": "Get available templates success."}, debug_output=False)
+            pro_conf_dir=os.getenv('PRO_CONF_DIR', os.getcwd())
+            conf_path = os.path.join(pro_conf_dir, "conf", "azureccprov_templates.json")
+            with open(conf_path, 'r') as json_file:
+                templates_json = json.load(json_file)
+            templates_json["message"] = "Get available templates success."   
+            return self.json_writer(templates_json, debug_output=False)
         except:
             logger.warning("Exiting Non-zero so that symphony will retry")
-            logger.exception("Could not get template_json")
-            sys.exit(1)
-
-    def generate_userdata(self, template):
-        ret = {}
-        
-        for key, value_array in template.get("attributes", {}).items():
-            if len(value_array) != 2:
-                logger.error("Invalid attribute %s %s", key, value_array)
-                continue
-            if value_array[0].lower() == "boolean":
-                ret[key] = str(value_array[1] != "0").lower()
-            else:
-                ret[key] = value_array[1]
-        
-        if template.get("customScriptUri"):
-            ret["custom_script_uri"] = template.get("customScriptUri")
-            
-        return ret
-        
-    def _parse_UserData(self, user_data):
-        ret = {}
-        
-        user_data = (user_data or "").strip()
-        
-        if not user_data:
-            return ret
-        
-        key_values = user_data.split(";")
-        
-        # kludge: this can be overridden either at the template level
-        # or during a creation request. We always want it defined in userdata
-        # though.
-        
-        for kv in key_values:
-            try:
-                key, value = kv.split("=", 1)
-                ret[key] = value
-            except ValueError:
-                logger.error("Invalid UserData entry! '%s'", kv)
-        return ret
-
-
-    def is_capacity_limited(self, bucket):
-        return False
+            logger.exception("Could not get templates_json")
+            sys.exit(1) 
     
-    def _max_count(self, nodearray_name, nodearray, machine_cores, bucket):
-        if machine_cores < 0:
-            logger.error("Invalid number of machine cores - %s.", machine_cores)
-            return -1
+    def generate_sample_template(self):
+        buckets = self.cluster.get_buckets()  
+        template_dict = {}
+        for bucket in buckets:
+            autoscale_enabled = bucket.software_configuration.get("autoscaling", {}).get("enabled", False)
+            if not autoscale_enabled:
+                print("Autoscaling is disabled in CC for nodearray %s" % bucket.nodearray)
+                continue    
+            if template_dict.get(bucket.nodearray) is None:
```

Comment:
This should trigger a warning level log entry, no? Someone has a nodearray with autoscale enabled, we should warn them and tell them they need to re-generate the static templates.

Reply:
meaning autoscaling disabled? so we could tell them this will not be included in generated template. Sometimes they want it disabled like persistent-execute nodes